rust/src/measurements/tulap/make_tulap.tex

```diff
@@ -0,0 +1,175 @@
+\documentclass{article}
+\input{../../lib.sty}
+
+\usepackage[
+        backend=biber,
+        style=authoryear-comp,
+        sorting=nyt,
+    ]{biblatex}
+\addbibresource{ref.bib} 
+
+\usepackage{thmtools}
+\usepackage{thm-restate}
+\newtheorem{thm}{Theorem}
+\newtheorem{defn}{Definition} 
+\newtheorem{prop}{Proposition}
+\newcommand{\mrm}[1]{\mathrm{#1}}
+
+
+\title{\texttt{fn make\_tulap}}
+\author{Yu-Ju Ku, Jordan Awan, Aishwarya Ramasethu, Michael Shoemate}
+\begin{document}
+
+\maketitle
+
+\contrib
+
+Proves soundness of \rustdoc{measurements/fn}{make\_tulap}.
+% in \asOfCommit{mod.rs}{f5bb719}.
+
+\texttt{make\_tulap} accepts parameters \texttt{input\_domain} and \texttt{input\_metric} specifying the input metric space.
+This consists of the space of non-null floating-point numbers, where distance is computed via the absolute distance.
+
+Additionally, the privacy parameters \texttt{epsilon} (of type \texttt{float}) and \texttt{delta} are fixed the privacy loss.
+The measurement assumes (but does not require) that input data (\texttt{arg}) follows the Binomial Distribution,
+and returns a sample from the Tulap distribution centered at \texttt{arg}.
+
+\begin{tcolorbox}
+    \begin{warning}[Code is not constant-time]
+        The implementation of \texttt{make\_tulap} uses procedures that are vulnerable to timing attacks. 
+    \end{warning}
+\end{tcolorbox}
+
+\subsection*{PR History}
+\begin{itemize}
+    \item \vettingPR{1126}
+\end{itemize}
+
+\section{Hoare Triple}
+
+\subsection*{Preconditions}
+\begin{itemize}
+    \item Variable \texttt{input\_domain} is of type \texttt{AtomDomain<f64>}.
+    \item Variable \texttt{input\_metric} is of type \texttt{AbsoluteDistance<f64>}.
+    \item Variable \texttt{epsilon} is of type \texttt{float}
+    \item Variable \texttt{delta} is of type \texttt{float}
+\end{itemize}
+
+\subsection*{Pseudocode}
+
+\lstinputlisting[language=Python,firstline=2]{./pseudocode/make_tulap.py}
+
+\subsection*{Postcondition}
+
+\validMeasurement{\texttt{(input\_domain, input\_metric, epsilon, delta)}}{\\ \texttt{make\_tulap}}
+
+\section{Proof}
+
+\begin{proof} 
+\hfill
+\begin{enumerate}
```

Comment:
Do we need to have an "enumerate" environment for this? Can't we just say,

\begin{Proof of Privacy:}{proof}

or something like that for a cleaner view and margins?